content/en/blog/2022/frontend-overhaul/index.md

```diff
@@ -0,0 +1,180 @@
+covering each of the following OTel-supported languages:
+
+- .NET
+- C++
+- Elixir
+- Go
+- Node.js
+- PHP
+- Python
+- Ruby
+- Rust
+
+Every microservice has a specific goal and can communicate with others by using
+a global gRPC definition. Persistent information is saved into a PostgresSQL
+database and there are outbound services that connect with third-party services
+to trigger events (such as confirmation emails). All of the microservices,
+including the front-end, are connected to the same OpenTelemetry collector
+instance, which uses Jaeger as one of the data stores for the tracing data.
+
+![OpenTelemetry Demo System Diagram](system-diagram.png)
+
+Prior to re-architecting, the front-end consisted of a Golang SSR app, which
+sent complete HTML to the browser for display. Every request and call redirected
+to the server so new information was shown.
+
+## Web App Styling Improvements, Theme Updates, and User Experience Redesign
+
+Before starting the development process, the front-end application wasn’t
+matching the theme that OpenTelemetry had been using in terms of colors,
+products, and overall user experience. In addition, the demo lacked a real
+front-end (browser side) application as the current implementation was a Go
+application.
+
+![OpenTelemetry Demo Old Front-end](old-design.png)
+
+The first task at hand was to bring the demo to the modern age by updating the
+design, color schemes, and user experience. Olly Babiak walked into the fray to
+help us achieve this by creating a modernized version of the application. It
+included an improved way to display the products landing page, an updated
+product details page, a mini cart, and a fully compatible mobile version of the
+application.
+
+![OpenTelemetry Demo New Front-end](new-design.png)
+
+Now we had an application design that would match the rest of the OpenTelemetry
+themes and colors and look more like the OpenTelemetry.io website.
+
+## Front-end Application Architecture Overhaul
+
+We worked on an initial proposal that included the following:
+
+- Framework and tooling (Scaffolding, I/O, styling, UI library)
+- Code Architecture and structure (Directories, coding patterns)
+- Instrumentation
+- Deployment & Distribution
+- Testing (E2E, unit test)
+
+This proposal was presented to the OpenTelemetry demo SIG during one of the
+weekly Monday meetings and we were given the green light to move ahead. As part
+of the changes, we decided to use [Next.js](https://nextjs.org/) to not only
+work as the primary front-end application but also to work as an aggregation
+layer between the front-end and the gRPC back-end services.
+
+![New Front-end Data Flow](data-flow.png)
+
+As you can see in the diagram, the application has two major connectivity
+points, one coming from the browser side (REST) to connect to the Next.js
+aggregation layer and the other from the aggregation layer to the back-end
+services (gRPC).
+
+## OpenTelemetry Instrumentation
```

Comment:
One thing that would be really great to have here, and I won't block on this but I still think it'd be helpful, would be an elaboration on how instrumentation is plugging into next.js. Specifically, I see that there's a next.js middleware implementation that creates spans. That's cool! But why is it there, and what does it do? How does it get used? I think that would be _very_ interesting to people who are also getting into tracing with frontends.

Reply:
Yep, that sounds neat! Maybe that could also make more sense as a readme or something in the demo repo itself. It's just something I got curious about since the post had me digging through code and it seemed cool but wasn't clear on the approach.